### PR TITLE
Custom error page

### PR DESCRIPTION
**Pull request by mzfr**
Updated clone.py to fetch error page style from original site.(fixes issue #66 )

_Not quite ready yet_

### CODE REVIEW

**Comment by viskey98** [comment]
Hi , @mzfr your 'custom error page' commit doesn't work and it has already been tested by a similar code done by me , it doesn't fetch the custom error page of the website , but a file stating "not found". And Iam currently working on that issue and i have completed it and currently discussing with @afeena about it, so u can let this issue (#66) aside and focus on the other ones :)

**Comment by mzfr** [comment]
@viskey98 Can you post the error that you get? I've tried it on a bunch of pages and it seems to work just fine.

**Comment by viskey98** [comment]
@mzfr  The commit you have made will just show a web-page with "NOT FOUND" written on it , when we request a page which is not found . But ,what we expect is : to show up a web-page with a custom 404 error page in the theme of the website which is cloned , for example look at this page (http://github.com/error_page) with the picture in the background and all the customised theme.

**Comment by mzfr** [comment]
@viskey98 That is very site-specific. Some sites try to be smart about the content they serve based on the headers they receive. In this particular case, Github uses the `Accept` header to send different content.

Running this command should give 'NOT FOUND':
``` curl 'https://github.com/error_page' ```

While running this will give the correct HTML page:
``` curl 'https://github.com/error_page' -H 'Accept: text/html' ```

I've pushed a fix for this!

File: clone.py [hunk 1]
```diff
@@ -145,6 +146,11 @@ def _make_filename(self, url):
             content_type = None
             try:
                 with aiohttp.Timeout(10.0):
+                    with aiohttp.ClientSession() as session:
+                        response = yield from session.get(current_url, headers={'Accept': 'text/html'})
```
Comment (viskey98) [inline clone.py]: use await syntax

File: clone.py [hunk 2]
```diff
@@ -145,6 +146,11 @@ def _make_filename(self, url):
...
+                        data = yield from response.read()
```
Comment (viskey98) [inline clone.py]: Please correct ur code by changing all "yield from" syntax into "await syntax"

File: clone.py [hunk 3]
```diff
@@ -169,6 +175,17 @@ def _make_filename(self, url):
...
+                            yield from self.new_urls.put(carved_url)
+
+    @asyncio.coroutine
+    def run(self):
+        yield from self.new_urls.put(self.root)
+        # Force 404 Page
+        yield from self.new_urls.put(self.error_page)
+        return (yield from self.get_body())
+
                         if carved_url.human_repr() not in self.visited_urls:
```
Comment (viskey98) [inline clone.py]: After the release , the whole code of this pr has messed up , please try to run this code and fix the errors :)

**Comment by glaslos** [comment]
Solved by #93

**Pull request closed without merging**